python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+                            "TARTS fx/fy data insufficient for donut %d: "
+                            "fx_list length=%d, fy_list length=%d. "
+                            "This may indicate a data mismatch between TARTS output and donut stamps.",
+                            i + 1, len(fx_list), len(fy_list)
+                        )
+                        row["extra_field_x"] = np.nan
+                        row["extra_field_y"] = np.nan
+                except AttributeError:
+                    # Fallback to calcFieldXY if TARTS fx/fy not available
+                    # Expected when TARTS doesn't provide field positions
+                    self.log.debug(
+                        "TARTS fx/fy not available for donut %d, using calcFieldXY fallback",
+                        i + 1
+                    )
+                    field_xy = extra.calcFieldXY()
+                    row["extra_field_x"] = field_xy[0] * u.deg
+                    row["extra_field_y"] = field_xy[1] * u.deg
+                # Use helper method for centroid calculation
+                centroid_x, centroid_y = self._get_centroid_for_stamp(extra, i)
+                row["extra_centroid_x"] = centroid_x * u.pixel
+                row["extra_centroid_y"] = centroid_y * u.pixel
+            else:
+                # extraStamp=None (padded when arrays have different lengths)
+                self.log.debug("No extra stamp available for donut %d", i + 1)
+                row["extra_field_x"] = np.nan
+                row["extra_field_y"] = np.nan
+                row["extra_centroid_x"] = np.nan
+                row["extra_centroid_y"] = np.nan
+
+
+            # Get quality metrics from metadata
+            for key in ["MAG", "SN", "ENTROPY", "FRAC_BAD_PIX", "MAX_POWER_GRAD", "FX", "FY"]:
+                for stamps, foc in [
+                    (intraStamps, "intra"),
+                    (extraStamps, "extra"),
+                ]:
+                    if (len(stamps) > 0 and key in stamps.metadata and
+                            i < len(stamps.metadata.getArray(key))):
+                        val = stamps.metadata.getArray(key)[i]
+                    else:
+                        val = np.nan
+                    row[f"{foc}_{key.lower()}"] = val
+
+            zkTable.add_row(row)
+
+        # Set metadata on the Zernike table
+        zkTable.meta = self.createZkTableMetadata()
+
+        # Debug: Log detector names in metadata
+        if "intra" in zkTable.meta and "det_name" in zkTable.meta["intra"]:
+            self.log.info("Zernike table intra det_name: '%s'", zkTable.meta["intra"]["det_name"])
+        if "extra" in zkTable.meta and "det_name" in zkTable.meta["extra"]:
+            self.log.info("Zernike table extra det_name: '%s'", zkTable.meta["extra"]["det_name"])
+
+        self.log.info("Created Zernike QTable with %d rows", len(zkTable))
+        return zkTable
+
+    def createZkTableMetadata(self) -> dict:
```

Comment:
I honestly think this function should be the same as the one in calcZernikesTask :
https://github.com/lsst-ts/ts_wep/blob/develop/python/lsst/ts/wep/task/calcZernikesTask.py#L337

Reply:
Please, could you clarify what are "the other differences"? Bryce has his unpaired task version here https://github.com/lsst-ts/ts_wep/pull/25/files which should have the same as you should need for yours